homeassistant/components/transmission/__init__.py

```diff
@@ -82,6 +90,14 @@ def refresh(event_time):
 
     track_time_interval(hass, refresh, scan_interval)
 
+    def add_torrent(service):
+        """Add new torrent to download."""
+        torrent = service.data[ATTR_TORRENT]
+        api.add_torrent(torrent)
```

Comment:
Do we need to check of this path is allowed with `hass.config.is_allowed_path`?

Reply:
Ah, yes. Correct!

Reply:
Only if that is a local file path

Reply:
Yeah, it can be a local file but doesn't have to be. I added some sanity checks that should be enough to cover what's needed.